File: src/main/kotlin/com/refined/stone/account/Account.java
package com.refined.stone.account;

import com.refined.stone.reaction.Reaction;
import kotlin.collections.ArrayDeque;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "account_id")
    private Long id;

    @Column
    private String email;

    @Column
    private String nickname;

    //one account to many reaction result
    @OneToMany(mappedBy = "account", cascade = CascadeType.ALL)
    List<Reaction> reactions = new ArrayList<>();

    //진영정보 CounterStrike, SuddenAttack
    @Enumerated(EnumType.STRING)
    private Team team;

    //NoArgsConstructor
    public Account(){

    }
    public Account(AccountRequestDto accountRequestDto){
        this.email= accountRequestDto.getEmail();
        this.nickname=accountRequestDto.getNickname();
        this.team = accountRequestDto.getTeam();
    }
}

File: backend/src/main/kotlin/com/refined/stone/account/Account.kt
package com.refined.stone.account

import com.refined.stone.reaction.Reaction
import javax.persistence.*


@Entity
class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "account_id")
    private val id: Long? = null

    @Column
    private var email: String? = null

    @Column
    private var nickname: String? = null

    //one account to many reaction result
    @OneToMany(mappedBy = "account", cascade = [CascadeType.ALL])
    var reactions: MutableList<Reaction> = ArrayList()

    //진영정보 CounterStrike, SuddenAttack
    @Enumerated(EnumType.STRING)
    private var team: Team? = null

    //NoArgsConstructor
    constructor() {}
    constructor(accountRequestDto: AccountRequestDto) {
        email = accountRequestDto.getEmail()
        nickname = accountRequestDto.getNickname()
        team = accountRequestDto.getTeam()
    }
}